### PR TITLE
fix dumping of hidden deps

### PR DESCRIPTION
**Pull request by boegel**
fix for #1350 

cc @ocaisa

### CODE REVIEW

**Comment by boegel** [comment]
@rjeschmi, @wpoely86: please review?

File: easybuild/framework/easyconfig/easyconfig.py [hunk 1]
```diff
@@ -196,6 +196,7 @@ def __init__(self, path, extra_options=None, build_specs=None, validate=True, hi
             self.validate(check_osdeps=build_option('check_osdeps'))
 
         # filter hidden dependencies from list of dependencies
+        self.orig_dependencies = None
```
Comment (wpoely86) [inline easybuild/framework/easyconfig/easyconfig.py]: why initialize to `None`?

Comment (boegel) [inline easybuild/framework/easyconfig/easyconfig.py]: mainly to silence pylint, it's mostly a style issue

I want to make the actual copy in `filter_hidden_deps`, where `dependencies` is being modified (potentially)

**Comment by wpoely86** [comment]
looks fine

**Comment by boegel** [comment]
Thanks for the review @wpoely86!

**Pull request merged**